File: 951-partition-array-into-disjoint-intervals/partition-array-into-disjoint-intervals.py
class Solution:
    def partitionDisjoint(self, nums: List[int]) -> int:
        lmax = nums[0]
        rmax = -1
        pos = 1
        for i in range(1,len(nums)):
            if lmax>nums[i]:
                pos = i+1
                lmax = max(lmax,rmax)
                rax = -1
            elif rmax<nums[i]:
                rmax = nums[i]
        return pos

        

File: 951-partition-array-into-disjoint-intervals/partition-array-into-disjoint-intervals.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int partitionDisjoint(std::vector<int>& nums) {
        int lMax = nums[0];
        int rMax = 0;
        int pos = 0;

        for (unsigned int i = 1; i < nums.size(); i++) {
            int currNum = nums[i];

            if (lMax > currNum) {
                pos = i;
                lMax = std::max(lMax, rMax);
                rMax = -1;
            } else{
                rMax = std::max(rMax, currNum);
            }
        }
        return pos+1;
    }
};
